addons/stock_account/models/account_invoice.py

```diff
@@ -16,7 +16,8 @@ def invoice_line_move_line_get(self):
         res = super(AccountInvoice, self).invoice_line_move_line_get()
         if self.company_id.anglo_saxon_accounting and self.type in ('out_invoice', 'out_refund'):
             for i_line in self.invoice_line_ids:
-                res.extend(self._anglo_saxon_sale_move_lines(i_line))
+                if not i_line.sale_line_ids.move_ids.mapped('move_line_ids.owner_id'):
```

Comment:
`sale_line_ids` is defined in `sale`, which doesn't depend on `stock_account`. The condition should be in `sale_stock`.